src/Service/CsvParser.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Service;
+
+use App\Factory\ProductFactory;
+use Symfony\Component\Finder\Finder;
+
+class CsvParser
+{
+    private $productsFileDir;
+    private $productsFileName;
+    private const FILE_EXT = '.csv';
+    /**
+     * @var Finder
+     */
+    private $finder;
```

Comment:
To prevent mistakes, it makes sense to use typed properties as we are using the high enough php version.